nix/overlay.nix

```
@@ -6,13 +6,29 @@ let
   contents = import ./contents.nix { nixpkgs = super; };
 
   inherit (super.lib.attrsets) mapAttrs;
-  inherit (import sources.gitignore { inherit lib; }) gitignoreSource;
+  inherit (import sources.gitignore { inherit lib; }) gitignoreFilter;
 
   ourOverrides = selfh: superh:
     let
 
+      customFilter = src:
+        let
+          # IMPORTANT: use the following let binding to memoize info about the
+          # Git directories.
+          gitIgnoredSrc = gitignoreFilter src;
+        in
+          path: type: gitIgnoredSrc path type;
```

Question:

With the comment, did you mean that the other alternative:
```suggestion
          path: type: gitignoreFilter src path type;
```

Will be un-memoized?

---

I don't really know, this was the comment from the example. I think there is a way to run `nix-build` with some timing information to see if there is indeed a difference.